File: 프로그래머스/Lv.1/최소직사각형.cs
//최소직사각형
//2021-12-25.
int solution(int[,] sizes)
{
    int xMax = 0, yMax = 0;
    for (int i = 0; i < sizes.GetLength(0); i++)
    {
        int x = sizes[i, 0];
        int y = sizes[i, 1];
        sizes[i, 0] = Math.Min(x, y);
        sizes[i, 1] = Math.Max(x, y);
        xMax = Math.Max(xMax, sizes[i, 0]);
        yMax = Math.Max(yMax, sizes[i, 1]);
    }

    return xMax * yMax;
}

File: 프로그래머스/코딩테스트 고득점 Kit/완전탐색/최소직사각형.cpp
// 최소직사각형
#include <bits/stdc++.h>
using namespace std;
int solution(vector<vector<int>> sizes)
{
    int x = 0;
    int y = 0;
    for (int i = 0; i < sizes.size(); i++)
    {
        int mx = max(sizes[i][0], sizes[i][1]);
        int mn = min(sizes[i][0], sizes[i][1]);

        x = max(x, mx);
        y = max(y, mn);
    }
    return x * y;
}